modules/compiler-core/src/internals/postprocess/AllOfTransformer.scala

```diff
@@ -228,6 +238,16 @@ private[compiler] object AllOfTransformer extends IModelPostProcessor {
     idToDef.toVector
   }
 
+  private def removeSelfReferentialMixin(definition: Definition): Definition =
+    definition match {
+      case struct @ Structure(id, _, _, hints) =>
+        struct.copy(hints = hints.filter {
+          case Hint.HasMixin(`id`) => false
+          case _                   => true
+        })
+      case other => other
+    }
```

Comment:
The downward propagation of hints led to scenarios in complex cases where a mixin would reference itself, causing a cyclical dependency and breaking the smithy spec.  I still need to figure out a minimum repro for this so I can write a test case for it.